### ISSUE TITLE
Cannot resolve BodyParserMiddleware namespace

### ISSUE THREAD

**Issue by schmanat** [adonisjs/core#2565]
Im setting up a new project as descripted in the documentation. After trying to start it witch ```node ace serve --watch``` I get the error message 

> IocLookupException: E_IOC_LOOKUP_FAILED: Cannot resolve "Adonis/Core/BodyParserMiddleware" namespace from the IoC Container

## Package version
```json
{
  "name": "coffeelist",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "build": "node ace build --production",
    "start": "node server.js",
    "dev": "node ace serve --watch"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^3.1.1",
    "adonis-preset-ts": "^2.1.0",
    "pino-pretty": "^4.7.1",
    "typescript": "~4.1",
    "youch": "^2.2.2",
    "youch-terminal": "^1.1.1"
  },
  "dependencies": {
    "@adonisjs/core": "^5.1.6",
    "@adonisjs/repl": "^1.1.6",
    "g": "^2.0.1",
    "npm": "^7.12.0",
    "proxy-addr": "^2.0.6",
    "reflect-metadata": "^0.1.13",
    "source-map-support": "^0.5.19"
  }
}
```

## Node.js and npm version
Node: 14.16.1
NPM: 7.9.0

## Sample Code (to reproduce the issue)
node ace serve --watch in the console 

## Expected outcome
a running enviroment

## Actual outcome
```
[ info ]  building project...


18  |
19  | An array of global middleware, that will be executed in the order they
20  | are defined for every HTTP requests.
21  |
22  */
23  Server.middleware.register([
24    'Adonis/Core/BodyParserMiddleware',
25  ])
26
27  /*
28  |--------------------------------------------------------------------------

  D:\wip\coffeelist\node_modules\@adonisjs\core\node_modules\@adonisjs\fold\build\src\Exceptions\IocLookupException.js:18

2 Ioc.lookupOrFail
  D:\wip\coffeelist\node_modules\@adonisjs\core\node_modules\@adonisjs\fold\build\src\Ioc\index.js:254

3 IocResolver.resolve
  D:\wip\coffeelist\node_modules\@adonisjs\core\node_modules\@adonisjs\fold\build\src\Resolver\index.js:80

4 MiddlewareStore.resolveMiddleware
  D:\wip\coffeelist\node_modules\@adonisjs\core\node_modules\@adonisjs\http-server\build\src\MiddlewareStore\index.js:62

5 MiddlewareStore.register
  D:\wip\coffeelist\node_modules\@adonisjs\core\node_modules\@adonisjs\http-server\build\src\MiddlewareStore\index.js:69

6 Module._compile
  D:\wip\coffeelist\node_modules\pirates\lib\index.js:99

[ warn ]  Underlying HTTP server died with "0 code"
[ info ]  watching file system for changes
```

## BONUS (a sample repo to reproduce the issue)
Not really necessary because its just the boilerplate

**Comment by RomainLanz** [comment]
Hey @schmanat! 👋 

It seems that you installed an old version of AdonisJS 5.
Please, ensure to add the `@latest` when running the init script.

```
npm init adonis-ts-app@latest
```

➡️ https://docs.adonisjs.com/guides/installation#creating-a-new-project

If you want to upgrade an application that was under the preview release, please, follow our upgrading process.

➡️ https://docs.adonisjs.com/releases/out-of-preview

**Comment by schmanat** [comment]
hey @RomainLanz 

thanks for your fast reply. I have done it your way (```npm init adonis-ts-app@latest```) but I got the same error as descriped above.

If I try to update an preview adonisjs project I got also the same errors:

```json
{
  "name": "backend",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "build": "node ace build --production",
    "start": "node server.js",
    "dev": "node ace serve --watch"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^5.1.1",
    "adonis-preset-ts": "^2.1.0",
    "pino-pretty": "^4.7.1",
    "typescript": "~4.1",
    "youch": "^2.2.1",
    "youch-terminal": "^1.1.0"
  },
  "dependencies": {
    "@adonisjs/auth": "^8.0.3",
    "@adonisjs/core": "^5.1.6",
    "@adonisjs/lucid": "^14.0.1",
    "@adonisjs/mail": "^7.1.1",
    "@adonisjs/repl": "^3.1.2",
    "@adonisjs/session": "^6.0.3",
    "@adonisjs/shield": "^7.0.2",
    "@adonisjs/view": "^6.0.1",
    "luxon": "^1.26.0",
    "phc-argon2": "^1.1.1",
    "proxy-addr": "^2.0.6",
    "reflect-metadata": "^0.1.13",
    "source-map-support": "^0.5.19",
    "sqlite3": "^5.0.2"
  }
}

```

**Comment by RomainLanz** [comment]
Concerning the error while migrating your application, the middleware definition [changed with the official release](https://docs.adonisjs.com/releases/out-of-preview#making-middleware-type-safe).

What template are you using when running `npm init adonis-ts-app@latest`?

**Comment by schmanat** [comment]
>What template are you using when running npm init adonis-ts-app@latest?

This is a good question, I dont define any template so i would say the default one :D. How can I find out which template is used?

Thanks for your hint with the middleware definition.
I created a new project ```npm init adonis-ts-app@latest my-project```. Then i opened the ```start/kernel.ts``` file and I saw the "old definiton"

```typescript
Server.middleware.register([
  'Adonis/Core/BodyParserMiddleware',
])
```

after changing to the new definition the app works. 

```typescript
Server.middleware.register([
 () => import('@ioc:Adonis/Core/BodyParser'),
])
```

One last question. How can i use the right template? Is there a cache which I should clear or something else?

Thanks again for your attention and help.

And sorry for my english :)
